test/FunctionalTests/Balancer/BalancerHelpers.cs

```diff
@@ -122,7 +123,9 @@ public static async Task<GrpcChannel> CreateChannel(ILoggerFactory loggerFactory
             var services = new ServiceCollection();
             services.AddSingleton<ResolverFactory>(new TestResolverFactory(resolver));
             services.AddSingleton<IRandomGenerator>(new TestRandomGenerator());
+            services.AddSingleton<IRandomGenerator>(new TestRandomGenerator());
```

Comment:
Why do we inject this twice here? It seems like we always resolve a single instance anyways?

Reply:
Copy/paste typo